fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -1798,14 +1798,43 @@ Version TagPartitionedLogSystem::getPeekEnd() const {
 		return std::numeric_limits<Version>::max();
 }
 
```

Comment:
It is not easy to follow this method.

I think this needs comments - what it is trying to do and how it is doing it.